### ISSUE TITLE
Proposal - Improvements to System.Environment.OSVersion (for Windows)

### ISSUE THREAD

**Issue by Akaion** [dotnet/corefx#40209]
It's well known that if an application isn't manifested for Windows 10, System.Environment.OSVersion.MajorVersion returns 6.2 (Windows 8). If I remember correctly this is due to changes in the Windows API function used.

There is, however, a way around this by using a different function.

The following code snippet will return the correct Major and Minor version number when `RtlGetVersion` is called.

```csharp
[DllImport("ntdll.dll", SetLastError = true)]
internal static extern uint RtlGetVersion(out OsVersionInfo versionInformation);

[StructLayout(LayoutKind.Sequential)]
internal struct OsVersionInfo
{
    private readonly uint OsVersionInfoSize;

    internal readonly uint MajorVersion;
    internal readonly uint MinorVersion;

    private readonly uint BuildNumber;

    private readonly uint PlatformId;

    [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 128)]
    private readonly string CSDVersion;
}
```

Note that as this is a native api call, the return value should actually be an NTSTATUS enum value.

**Comment by danmosemsft** [comment]
Windows did this to stop breaking apps that unwisely check version number rather than calling  capability API. I am not sure we should work against that. Why do you need it? If you do, it is probably easiest to get from the registry.

**Comment by jkotas** [comment]
> an application isn't manifested for Windows 10

net core 3.0 apps are manifested for Windows 10 by default. It was a bug that was fixed for net core 3.0 - earlier net core versions did not have the right manifest.

**Comment by Akaion** [comment]
Right, well I guess if core 3.0 apps are automatically manifested for Windows 10 then that sort of solves the issue. I was just a bit concerned regarding the fact that an API said it did one thing, but only if the app was manifested (required me to go digging through documentation, and looking up the API calls.)
